packages/beacon-node/src/chain/bls/multithread/jobItem.ts

```diff
@@ -70,6 +73,42 @@ export function jobItemWorkReq(job: JobQueueItem, format: PointFormat, metrics:
       // and not a problem in the near future
       // this is monitored on v1.11.0 https://github.com/ChainSafe/lodestar/pull/5912#issuecomment-1700320307
       const timer = metrics?.blsThreadPool.signatureDeserializationMainThreadDuration.startTimer();
+
+      // adding verification randomness is swig specific. must not attempt with herumi until
+      // @chainsafe/bls is updated to support it with herumi
+      if (job.opts.addVerificationRandomness) {
+        const pkPoint = new swigBindings.blst.P1();
+        const sigPoint = new swigBindings.blst.P2();
+        for (let i = 0; i < job.sets.length; i++) {
+          const randomness = randomBytesNonZero(8);
+          // cast to unknown here because the blst-native version of the bls library extends the
+          // PublicKey from the blst library, but the herumi version does not so the interface does
+          // not show that this is possible
+          const {value} = job.sets[i].publicKey as unknown as blst.PublicKey;
+          let dup: swigBindings.P1;
+          if (typeof (value as swigBindings.P1_Affine).to_jacobian === "function") {
```

Comment:
Is there anything that could be done on `bls` side to remove:
* the blst specific cast
* the `to_jacobian` check
* the `blst.Signature` usage

Might be worth creating an issue in the `bls` repo?

Also, what does that mean if this is executed in `env` not running native `bls` security-wise? (retorical question for now as there is no such env AFAIK)

Reply:
duplicate. resolving.